openfe/tests/setup/test_lomap_scorers.py

```diff
@@ -0,0 +1,251 @@
+    r1 = Chem.AddHs(Chem.MolFromSmiles(base))
+    r2 = Chem.AddHs(Chem.MolFromSmiles(other))
+    # add 2d coords to stop Lomap crashing for now
+    for r in [r1, r2]:
+        Compute2DCoords(r)
+    m1 = openfe.setup.SmallMoleculeComponent.from_rdkit(r1)
+    m2 = openfe.setup.SmallMoleculeComponent.from_rdkit(r2)
+
+    mapper = openfe.setup.LomapAtomMapper()
+    mapping = next(mapper.suggest_mappings(m1, m2))
+    score = lomap_scorers.heterocycles_score(mapping)
+
+    assert score == 0 if not hit else score == 1 - math.exp(-0.4)
+
+
+# test individual scoring functions against lomap
+SCORE_NAMES = {
+    'mcsr': 'mcsr_score',
+    'mncar': 'mncar_score',
+    'atomic_number_rule': 'atomic_number_score',
+    'hybridization_rule': 'hybridization_score',
+    'sulfonamides_rule': 'sulfonamides_score',
+    'heterocycles_rule': 'heterocycles_score',
+    'transmuting_methyl_into_ring_rule': 'transmuting_methyl_into_ring_score',
+    'transmuting_ring_sizes_rule': 'transmuting_ring_sizes_score'
+}
+IX = itertools.combinations(range(8), 2)
+
+
+@pytest.mark.parametrize('params', itertools.product(SCORE_NAMES, IX))
+def test_lomap_individual_scores(params,
+                                 lomap_basic_test_files):
+    scorename, (i, j) = params
+    mols = sorted(lomap_basic_test_files.items())
+    _, molA = mols[i]
+    _, molB = mols[j]
+
+    # reference value
+    lomap_version = getattr(lomap.MCS(molA.to_rdkit(),
+                                      molB.to_rdkit()), scorename)()
+
+    # longer way
+    mapper = openfe.setup.LomapAtomMapper()
+    mapping = next(mapper.suggest_mappings(molA, molB))
+    openfe_version = getattr(lomap_scorers, SCORE_NAMES[scorename])(mapping)
+
+    assert lomap_version == pytest.approx(1 - openfe_version), \
+           f"{molA.name} {molB.name} {scorename}"
+
+
+# full back to back test again lomap
+def test_lomap_regression(lomap_basic_test_files_dir,  # in a dir for lomap
+                          lomap_basic_test_files):
+    # run lomap
+    dbmols = lomap.DBMolecules(lomap_basic_test_files_dir)
+    matrix, _ = dbmols.build_matrices()
+    matrix = matrix.to_numpy_2D_array()
+
+    assert matrix.shape == (8, 8)
+
+    # now run the openfe equivalent
+    # first, get the order identical to lomap
+    smallmols = []
+    for i in range(matrix.shape[0]):
+        nm = dbmols[i].getName()
+        smallmols.append(lomap_basic_test_files[nm[:-5]])  # - ".mol2"
+
+    mapper = openfe.setup.LomapAtomMapper()
+    scorer = lomap_scorers.default_lomap_score
+    scores = np.zeros_like(matrix)
+    for i, j in itertools.combinations(range(matrix.shape[0]), 2):
+        molA = smallmols[i]
+        molB = smallmols[j]
+
+        mapping = next(mapper.suggest_mappings(molA, molB))
+        score = scorer(mapping)
+
+        scores[i, j] = scores[j, i] = score
+    scores = 1 - scores
+    # fudge diagonal for comparison
+    for i in range(matrix.shape[0]):
+        scores[i, i] = 0
```

Comment:
This is the code snippet that recreates lomap scoring matrix with our machinery

Reply:
If our network object already has scores associated, can we just use nx for this? https://networkx.org/documentation/stable/reference/generated/networkx.convert_matrix.to_numpy_array.html